Météo-france sensor : 1 hour rain forecast

victorcerutti opened the pull request:
## Description:
Météo France provides a convenient rain forecast that is pretty accurate.
This sensor gives the time to next rain forecasted.

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#<home-assistant.io PR number goes here>
home-assistant/home-assistant.io#6938

## Example entry for `configuration.yaml` (if applicable):
```yaml
sensor:
  - platform: meteo_france
    location_id: 75101
    name: Paris 1er
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

victorcerutti committed:
🌧 Meteo France rain forecast

victorcerutti committed:
🌧 Meteo France rain forecast

victorcerutti committed:
🌧 Meteo France rain forecast

victorcerutti committed:
do not fail on wrong config

victorcerutti committed:
do not fail on wrong config

victorcerutti committed:
do not fail on wrong config

victorcerutti committed:
Update name of sensor class

victorcerutti committed:
Update name of sensor class

victorcerutti committed:
Update name of sensor class

victorcerutti committed:
do not show sensor if not working

victorcerutti committed:
do not show sensor if not working

victorcerutti committed:
do not show sensor if not working

victorcerutti committed:
Update .coveragerc

victorcerutti committed:
Update .coveragerc

victorcerutti committed:
Update .coveragerc

victorcerutti committed:
ability to submit insee location code without final 0 needed by meteo-france

victorcerutti committed:
ability to submit insee location code without final 0 needed by meteo-france

victorcerutti committed:
ability to submit insee location code without final 0 needed by meteo-france

victorcerutti committed:
Lynting

victorcerutti committed:
Lynting

victorcerutti committed:
Lynting

victorcerutti committed:
more lynting

victorcerutti committed:
more lynting

victorcerutti committed:
more lynting

emontnemery committed:
Add binary_sensor support to RFlink (#17146)

* Add binary_sensor support to RFlink

* Add support for aliases

* Fix review comments

* Refactor, add tests

* Review comments

* Review comments

* Review comments

* Review comments

victorcerutti committed:
update comment

victorcerutti committed:
update comment

victorcerutti committed:
update comment

victorcerutti committed:
block comment linting

victorcerutti committed:
block comment linting

victorcerutti committed:
block comment linting

victorcerutti committed:
reducing length of long lines

victorcerutti committed:
reducing length of long lines

victorcerutti committed:
reducing length of long lines

amelchio committed:
Fix flux switch update interval (#17458)

victorcerutti committed:
linting

victorcerutti committed:
linting

victorcerutti committed:
linting

victorcerutti committed:
Merge branch 'meteofrance' of https://github.com/victorcerutti/home-assistant into meteofrance

mvn23 committed:
Add sensor support to opentherm_gw (#17314)

* Add OpenTherm Gateway sensor platform.

* Add OTGW_ variables to list of supported sensors.

* Order imports.

* Add OpenTherm Gateway binary sensor support.

* Revert "Add OpenTherm Gateway binary sensor support."

This reverts commit 115acaa912f4f9ff47408a04f4a770bebd071f26.

* Import COMP_SENSOR from sensor component rather than defining it.

* Update opentherm_gw sensor platform docs url.

* Update dependency to v0.2b1
Old version had incorrect variable names for some of the sensors

* Update requirements_all.txt

* Address review findings.

* Update .coveragerc

jjlawren committed:
Add missing await for coroutine (#17609)

thomasloven committed:
Tuya light icon fix (#17605)

* Fix for tuya light icons going too bright

* Make sure other values aren't strings either

thomasloven committed:
Revert "De-syncing binary_sensor.ping (#17056)" (#17606)

This reverts commit 11d5671ee0a4dbf98c5e9f58a2dec85bdb35f4d0.

ehendrix23 committed:
Add doorsense sensor for August 3rd Gen Smart Lock Pro (#17299)

* Add doorsense sensor for August 3rd Gen Smart Lock Pro

Add a binary sensor to August for the August 3rd Gen Smart Lock Pro doorsense sensor.

This is a re-do from PR 17116 https://github.com/home-assistant/home-assistant/pull/17116 that I closed due to rebase issue on my end.

* Changed to use snjoetw provided code

Going through the py-august I found that snjoetw had provided updated versions for the august component (august.py and binary_sensor/august.py) to include DoorSense sensor.
Changed what I did to to what snjoetw provided instead as he split it into 2 classes; much cleaner I think.

I modified his coding with:
   Fixes that were done to the August component and not part of the coding snjoetw provided.
   Added the debug logging improvement I had done in the code.

Note, fix I committed earlier for lock atribute (lock/august.py) is thus still the same.

* Reverted change from add_device to add_entities

Missed an item when merging snjoetw's code with current. Fixed.

* Updated call from add_devices to add_entities as well

Updated the call from add_devices to add_entities.

* Fixed permissions on files

Fixed permissions on components/august.py and binary_snesor/august.py

* Changed if/else to if/continue

Changed logic so that if the door sensor state is unknown during initalization the debug log is written and then continue the loop instead of using if/else logic.

* Added available property for Door Sensor

Added the available property for the Door Sensor and setting it to False if a status unknown is received.

* Updated setting self._available

Changed line for setting self._available to what Martin provided. Much  more efficient to read. :-)

nhorvath committed:
Bump thermoworks version to fix conflict from upstream pyrebase sseclient (#17620)

StevenLooman committed:
Upgrade async_upnp_client to 0.12.7 (#17601)

partofthething committed:
Upgrade gstreamer-player to 1.1.2 (#17568)

* Upgrade gstreamer-player to 1.1.2

* Updated requirements for gstreamer-player properly.

cdce8p committed:
Add water_heater support to HomeKit (#17614)

* Homekit add support for water_heater
* Added tests

Commit:
Add Elk-M1 sensor platform (#17342)

* Initial sensor version.

* Add Elk-M1 sensor services. Initial version of services.yaml.

* Fix lint errors.

* Fix PR comments.

* fix PR comment

* Fix PR comment

* Fix PR comments

* fix

cdce8p committed:
Homekit component cleanup (#17627)

* hass.async_add_executor_job
* Fix accessories.run -> async_track_state_change
* Fixed media_player test
* Flags are now local vars
* consistent use of " and '

dshokouhi committed:
Use cached robot serial for Neato update (#17633)

* Use cached robot serial for neato update

* Hound

Kane610 committed:
deCONZ - Add support for Xiaomi window covers (#17337)

Add support for Xiaomi window covers

Debaru committed:
Fix linky sensor login error (#17110)

* Fix linky sensor login error

* Make platform fail-safe

Adding following enhancements:
* Make sure the platform loads correctly by making the first API request in setup_platform.
* Close the session after each API call.
* Use timeout parameter everywhere.

* Fix Hound CI error: line too long.

* Update pylinky library

* Remove LinkyClient from update()

MartinHjelmare commented on homeassistant/components/sensor/meteo_france.py:
```diff
@@ -0,0 +1,148 @@
+    def update(self):
+        """Get the latest data from BOM."""
+        try:
+            result = requests.get(self._build_url(), timeout=10).json()
```
Please extract service specific I/O and data parsing to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services

thinkl33t committed:
Fix mqtt light brightness slider (#17075)

* Enable brightness slider for RGB

If we are using RGB with no brightness topic, the brighness slider
should still be visible, as we can scale the RGB amount to give us the
brightness.

* Output RGB scaled by brightness

If we are outputting to an RGB device, but do not have a dedicated
brightness topic set, when the brightness slider is changed, we should
output the current colour's HS, with the V coming from the brightness
slider.

* Brightness from RGB when we're not using a brightness topic

When we aren't using a brightness topic, set the brightness slider based
on the received value from an RGB -> HSV conversion.

* Test for new brightness state scaled by RGB

This adds a test to make sure the brightness stored in the state is
being computed correctly from the RGB value when a dedicated brightness
topic is not set.

* Changes from review

Fixes formatting of supported features flags, and checks HS colour
hasn't been set when operating in RGB-only mode

* Set optimistic brightness correctly in rgb mode

When we're using rgb mode to set the brightness, we want to set
optimistic brightness if:

we are running in optimistic mode
OR
the brightness state topic isn't set and we have a brightness command topic
OR
the rgb state topic isn't set and we don't have a brightness command topic

* Add test for turn_on in RGB brightness mode

mvn23 committed:
Add opentherm_gw binary sensor support (#17625)

* Add OpenTherm Gateway binary sensor support.

* opentherm_gw binary_sensor platform does not need polling.

victorcerutti commented on homeassistant/components/sensor/meteo_france.py:
```diff
@@ -0,0 +1,148 @@
+                    return self._time_to_rain
+                self._time_to_rain += 5
+            self._unit = ""
+            return "No rain"
```
I understand but what is displayed here is the time to the next expected rain.
If it's already raining or if it will start in the next 5 minutes, then the value returned is 0.
What value should be returned if there is no rain for the next 60 minutes then ? Certainly not zero, and returning an int like 60 or 100 doesn't make much sense either.

victorcerutti commented:
> We should also consider to make it a `weather` platform. Usually the rain forecast is only one element of a meteorological sensor.

@fabaff : I agree, but there is no API for the other weather datas.
I can easily scrape them, but is it an allowed practice to rely on scrapping for a weather component ?
I know this is only my point of view, but weather datas from other sources already in hass are already quite reliable.
This 1hr raining forecast, on the contrary, is rather unique and fits perfectly the need to know if I should close an opened windows for exemple.

Regarding all other comments, thanks for the advices. I will update my code.

raymondelooff committed:
Upgrade dsmr_parser to 0.12 (#17634)

guillaume1410 committed:
Removing ryobi gdo (#17637)

Danielhiversen committed:
Remove ryobi from .coveragerc (#17647)

balloob committed:
Reconnect if sub info comes in that is valid again (#17651)

emontnemery committed:
Very minor cleanup of RFLink components (#17649)

Danielhiversen committed:
available to switchmate (#17640)

fabaff committed:
Upgrade async_timeout to 3.0.1 (#17655)

fabaff committed:
Upgrade holidays to 0.9.8 (#17656)

soraxas committed:
Add support for Dyson Hot+Cool Fan as a climate device (#14598)

* Added support for dyson hot+cool fan as climate device

* Removed decimal place in kelvin units conversion

Minor edits to be consistent with Dyson's internal conversion of temperature from kelvin to celsius. It does not include decimal place to convert between kelvin and celsius.

* made changes according to comments

* Refactored target temp logics, fixed enum issues

* changed name of component to entity

* removed temperature conversion for min/max property

* changed back to 644 permission

* added extra tests for almost-all coverage

* changed assert method to avoid lack of certain method in py35

* added test_setup_component

* shorten line length

* fixed mock spec and added checking of message listener is called

* added doc string and debug msg

* shorten line length

* removed pending target temp

MatMaul committed:
Update pynetgear to 0.5.0 (#17652)

lukiffer committed:
Bump pyarlo==0.2.2 (#17673)

* Bump pyarlo to 0.2.2, fixes #17427

* Increase log level for refresh message to clear up logs

Commit:
Add new rtorrent sensor (#17421)

* New rtorrent sensor

* Fix lint issue

* Fix another lint issue

* Fix pylint issue

* how many python linters do you guys use

* Cleanup code

* python linting

* newline

balloob committed:
Bump frontend to 20181021.0

balloob committed:
Update translations

oblogic7 committed:
Update flux library version (#17677)

bramkragten committed:
Add lovelace websocket get and set card (#17600)

* Add ws get, set card

* lint+fix test

* Add test for set

* Added more tests, catch unsupported yaml constructors

Like !include will now give an error in the frontend.

* lint

Danielhiversen committed:
Mill, support more heater types (#17676)

* mill, suport more heater types

* mill requirements

quazzie committed:
Add missing hangouts data/image to notify service (#17576)

* add missing hangouts image_file/url to notify services

Missed adding support for hangouts image to notify service

* default in schema

exxamalte committed:
Geo location trigger added (#16967)

* zone trigger supports entity id pattern

* fixed lint error

* fixed test code

* initial version of new geo_location trigger

* revert to original

* simplified code and added tests

* refactored geo_location trigger to be based on a source defined by the entity

* amended test cases

* small refactorings

manuel-jrs committed:
Fix Mexican Spanish identifier (#17674)

Mexican Spanish identifier is "es-MX" instead of "en-MX".

tmonck committed:
Add readthedoc.yml file to specify the version of python to run during documentation building. (#17642)

doudz committed:
All supported domains should be exposed by default (#17579)

According to documentation, all supported domains should be exposed by default
https://www.home-assistant.io/components/google_assistant/#expose_by_default

bachya committed:
Updated simplisafe-python to 3.1.13 (#17696)

amelchio committed:
Update limitlessled to 1.1.3 (#17703)

fabaff committed:
Upgrade Sphinx to 1.8.1 (#17701)

Commit:
Away mode temperature fix for generic thermostat (#17641)

* Resolves /home-assistant/home-assistant#17433

Away mode temperature issue fix for generic_thermostat

* Debug messages removed from generic_thermostat.py

* Test for repeat away_mode set

Test for fix of generic thermostat issue when away_mode was set several times in a row.

* Code style fix in generic_thermostat

* Remove blank line in the end of generic_thermostat

* Fix style

JaxomCS committed:
Bug fix for clicksend (#17713)

* Bug fix

Current version causes 500 error since it is sending an array of from numbers to ClickSend. Changing the from number to 'hass' identifies all messages as coming from Home Assistant making them more recognisable and removes the bug.

* Amendment

Changed it to use 'hass' as the default instead of defaulting to the recipient which is the array. Would have worked if users set their own name but users who were using the default were experiencing the issue.

* Added DEFAULT_SENDER variable

nickovs committed:
Fixed issue #16903 re exception with multiple simultanious writes (#17636)

Reworked tests/components/emulated_hue/test_init.py to not be
dependent on the specific internal implementation of util/jsonn.py

kennedyshead committed:
Async version for asuswrt (#17692)

* Testing async data for asuswrt

* Moved to lib

rohankapoorcom committed:
Migrate Mailgun to use the webhook component (#17464)

* Switch mailgun to use webhook api

* Generalize webhook_config_entry_flow

* Add tests for webhook_config_entry_flow

* Add tests for mailgun

* Remove old mailgun file from .coveragerc

* Refactor WebhookFlowHandler into config_entry_flow

* Remove test of helper func from IFTTT

* Lint

d0ugal committed:
Only strip from the bluetooth name if it isn't None (#17719)

This prevents the following traceback that will otherwise occur.

    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/concurrent/futures/thread.py", line 56, in run
        result = self.fn(*self.args, **self.kwargs)
      File "/usr/local/lib/python3.6/site-packages/homeassistant/components/device_tracker/bluetooth_le_tracker.py", line 107, in update_ble
        see_device(address, devs[address], new_device=True)
      File "/usr/local/lib/python3.6/site-packages/homeassistant/components/device_tracker/bluetooth_le_tracker.py", line 47, in see_device
        see(mac=BLE_PREFIX + address, host_name=name.strip("\x00"),
    AttributeError: 'NoneType' object has no attribute 'strip'

kennedyshead committed:
Just use debug instead of error if the binary_sensor does not get data (#17720)

balloob committed:
Update frontend to 20181023.0

balloob committed:
Update translations

Danielhiversen committed:
clean up clicksend (#17723)

ehendrix23 committed:
Set available property (#17706)

Will set the available property to False if unable to communicate with August lock or doorbell.
HTTP request errors (i.e. timeout, connection error, HTTP error) will not result in traceback. Instead an error will be logged.

kennedyshead committed:
Removes re-init (#17724)

foxel committed:
Don't call off_delay_listener if not needed (#17712)

Don't call off_delay_listener if 'OFF' is actually received
Moved `off_delay_listener` to be defined once

fabaff committed:
Rename readthedocs file (#17718)

* Rename file

* Add requirements file

jxwolstenholme committed:
Add device tracking for the BT Smart Hub router (#17158)

* Support for BT smarthub router device tracking

* Update requirements_all.txt for bt_smarthub device tracker

* Added bt_smarthub.py exclusion

* Update .coveragerc

* Update bt_smarthub.py

* Update bt_smarthub.py

* Update bt_smarthub.py

* Update bt_smarthub.py

* Fix linting issues

* Update bt_smarthub.py

* Update bt_smarthub.py

* Update bt_smarthub.py

* Update bt_smarthub.py

* Update bt_smarthub.py

* Update bt_smarthub.py

* Update bt_smarthub.py

* Update bt_smarthub.py

* Update .coveragerc

* Update bt_smarthub.py

* Minor changes

* Update bt_smarthub.py

* Update bt_smarthub.py

scop committed:
Upgrade pytest to 3.9.2 (#17736)

scop committed:
Upgrade mypy to 0.641 (#17734)

bramkragten committed:
Lovelace ws: add card (#17730)

* Change set to update

* Add 'add card'

* Woof.

StevenLooman committed:
IGD review fixes (#17400)

* Fix discovery-dependency for upnp

* Only delete port mappings on EVENT_HOMEASSISTANT_STOP + refactoring MockDevice

* Call and store local_ip from async_setup

* Don't depend on http-component

* Remove discovery dependency

Commit:
Elk-M1 climate (#17679)

* Initial climate for Elk-M1.

* Tidy

* fix hound error

* fix hound error

cgarwood committed:
Add cover to supported platforms (#17725)

balloob committed:
Pytest tests (#17750)

* Convert core tests

* Convert component tests to use pytest assert

* Lint 🤷‍♂️

* Fix test

* Fix 3 typos in docs

rohankapoorcom committed:
Remove webhook_id from yaml config for mailgun (#17732)

Hedda committed:
Update zha __init__.py to reflect new Zigbee name stylization by the Zigbee Alliance (#17751)

Zigbee Alliance has changed their stylized writing (and logo) of the Zigbee name from “ZigBee” to “Zigbee”, as in they are no longer writing Zigbee with a capital “B” in the middle of the name.

Danielhiversen committed:
Clean up clicksend_tts (#17749)

* Clean up clicksend_tts

* style

fabaff committed:
Remove day (fixes #17741) (#17743)

balloob committed:
Convert auth websocket commands to use async_response decorator (#17755)

fabaff committed:
Update name (fixes #17752) (#17756)

victorcerutti committed:
Merge branch 'meteofrance' of https://github.com/victorcerutti/home-assistant into meteofrance

victorcerutti committed:
Update météo-france platform

Now work with pypi package and monitored conditions

victorcerutti commented:
Sorry I screwed up this PR. I'm closing it and submitting a new one for the same platform
